src/mlpack/core/data/load_impl.hpp

```diff
@@ -164,13 +164,25 @@ bool Load(const std::string& filename,
   if (loadType != FileType::HDF5Binary)
   {
     if (loadType == FileType::CSVASCII)
-      success = loader.LoadNumericCSV(matrix, stream);
+    {
+      arma::field<std::string> headers;
+      if (transpose)
+        success = matrix.load(arma::csv_name(filename, headers,
+              arma::csv_opts::trans), arma::csv_ascii);
+      else
+        success = matrix.load(arma::csv_name(filename, headers),
+            arma::csv_ascii);
+    }
     else
       success = matrix.load(stream, ToArmaFileType(loadType));
   }
   else
     success = matrix.load(filename, ToArmaFileType(loadType));
 
+  // Now transpose the matrix, if necessary.
+  if (transpose && loadType != FileType::CSVASCII)
+    success = inplace_transpose(matrix, fatal);
```

Comment:
I moved it here because of two reasons:

1. The return of this function is not being inspected, and not being handled.
2. This is making the Log not compatible with the current situation, and it should only be shown if we have really done transpose and not assuming we are going to do it.
3. A better way to do it is to handle/catch the loading error before transposing and not just print the matrix size.

Reply:
You are right that the success of the `inplace_transpose()` operation wasn't being checked directly, but, it is the return value of `data::Load()` so the user could check it.  I don't mind adding an extra block to check it and printing that load (or transpose) failed.

But, by moving the check up here, we no longer check the output of the actual load operation (`matrix.load()`).  I might suggest restructuring the whole code like this so that we successfully capture errors in both cases and only transpose when needed:

```c++
if (loadType == FileType::CSVASCII)
{
  // the CSV loading code goes here
}
else
{
  if (loadType != FileType::HDF5Binary)
    success = matrix.load(stream, ToArmaFileType(loadType));
  else
    success = matrix.load(filename, ToArmaFileType(loadType));

  if (success && transpose)
    success = inplace_transpose(X);
}

if (!success)
{
  // print that loading failed
}
```

But: I also noticed while writing this comment that `inplace_transpose()` doesn't really do much above and beyond what Armadillo already does.  I think it would be fine to remove that function and just use `X = X.t()`; Armadillo will automatically do this in-place where possible.  Maybe it was useful many years ago, but at least now, I don't know that it is.